### PR TITLE
feat(MeshMetric): implement OpenTelemetry API for MeshMetric

### PR DESCRIPTION
**Pull request by Automaat**
Fixes: #8847 

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/policies/meshmetric/api/v1alpha1/validator.go [hunk 1]
```diff
@@ -77,6 +79,12 @@ func validateBackend(backends *[]Backend) validators.ValidationError {
 			} else {
 				verr.Add(validators.ValidatePort(path.Field("port"), backend.Prometheus.Port))
 			}
+		case OpenTelemetryBackendType:
+			if backend.OpenTelemetry == nil {
+				verr.AddViolationAt(path.Field("openTelemetry"), validators.MustBeDefined)
+			} else if !govalidator.IsURL(backend.OpenTelemetry.Endpoint) {
```
Comment (slonka) [inline pkg/plugins/policies/meshmetric/api/v1alpha1/validator.go:85]: just wanted to check if we have the same validation for MAL and there we only check if it's defined - https://github.com/kumahq/kuma/blob/fecb3cf9cbb4d2552d8b24aa676ef1446095d036/pkg/plugins/policies/meshaccesslog/api/v1alpha1/validator.go#L116-L120 @lobkovilya - any reason for that? I think checking `govalidator.IsURL` is safer.

File: pkg/plugins/policies/meshmetric/api/v1alpha1/validator_test.go [hunk 2]
```diff
@@ -108,6 +108,44 @@ default:
...
+  kind: MeshService
+  name: svc-1
+default:
+  backends:
+    - type: OpenTelemetry
+      openTelemetry:
+        endpoint: "asdasd123"
+`),
+		resources.ErrorCase(
+			"invalid regex",
```
Comment (slonka) [inline pkg/plugins/policies/meshmetric/api/v1alpha1/validator_test.go:132]:
```suggestion
			"undefined openTelemetry backend when type is OpenTelemetry",
```

File: pkg/plugins/policies/meshmetric/api/v1alpha1/validator_test.go [hunk 3]
```diff
@@ -108,6 +108,44 @@ default:
...
+mesh: mesh-1
+name: metrics-1
+targetRef:
+  kind: MeshService
+  name: svc-1
+default:
+  backends:
+    - type: OpenTelemetry
+      prometheus:
+        endpoint: "asdasd123"
```
Comment (slonka) [inline pkg/plugins/policies/meshmetric/api/v1alpha1/validator_test.go:148]: nitpick:

```suggestion
        port: 5670
```

**Review by slonka — APPROVED** [review]
Mostly small stuff.

File: pkg/plugins/policies/meshmetric/api/v1alpha1/validator_test.go (hunk 3 above)
Comment (slonka) [reply pkg/plugins/policies/meshmetric/api/v1alpha1/validator_test.go:148]: nvm

File: pkg/plugins/policies/meshmetric/api/v1alpha1/validator.go (hunk 1 above)
Comment (Automaat) [reply pkg/plugins/policies/meshmetric/api/v1alpha1/validator.go:85]: I've created issue for this: https://github.com/kumahq/kuma/issues/8884

**Pull request merged**